### PR TITLE
Add option to create a bucket on a specific region

### PR DESCRIPTION
**Pull request by nimrod-becker**
Add an option to create a bucket on a specific region. 
When running against S3, you can specify the specific region for the bucket.

### CODE REVIEW

**Comment by mykaul** [comment]
I'm not sure the default should be set and not None. If it's not None, then it should be used.

**Comment by nimrod-becker** [comment]
us-east-1 is the default of the sdk if you don't provide any region. This just makes it more explicit and visible.

File: ocs_ci/ocs/resources/noobaa.py [hunk 1]
```diff
@@ -48,16 +48,22 @@ def __init__(self):
...
+            region: Name of the region for the bucket to be created on (defualt is us-east-1, as the SDK default)
 
         Returns:
             s3.Bucket object
 
         """
-        return self.s3_resource.create_bucket(Bucket=bucketname)
+        return self.s3_resource.create_bucket(
+            Bucket=bucketname,
+            CreateBucketConfiguration={
```
Comment (ebenahar) [inline ocs_ci/ocs/resources/noobaa.py]: Can we pass multiple regions here? Seems like 'LocationConstraint' key can get only a string accroding to https://docs.aws.amazon.com/sdkforruby/api/Aws/S3/Types/CreateBucketConfiguration.html

Comment (ebenahar) [reply ocs_ci/ocs/resources/noobaa.py]: Sorry, this one:
https://boto3.amazonaws.com/v1/documentation/api/latest/guide/s3-example-creating-buckets.html

Comment (nimrod-becker) [reply ocs_ci/ocs/resources/noobaa.py]: A bucket is always associated with 1 region only.

File: ocs_ci/ocs/resources/noobaa.py [hunk 2]
```diff
@@ -48,16 +48,22 @@ def __init__(self):
             aws_secret_access_key=self.access_key
         )
 
-    def s3_create_bucket(self, bucketname):
+    def s3_create_bucket(self, bucketname, region='us-east-1'):
```
Comment (ebenahar) [inline ocs_ci/ocs/resources/noobaa.py]: The default region should be taken from the env vars:
```python
from ocs_ci.framework import config

self.region = config.ENV_DATA['region']
```

File: ocs_ci/ocs/resources/noobaa.py (hunk 1 above)
Comment (ebenahar) [reply ocs_ci/ocs/resources/noobaa.py]: It's clear now, after an offline discussion.
Thanks @nimrod-becker

File: ocs_ci/ocs/resources/noobaa.py (hunk 2 above)
Comment (ebenahar) [reply ocs_ci/ocs/resources/noobaa.py]: @MeridianExplorer @parikshithb Please change the region here to be as written above, in a following PR.
Thanks

Comment (RazTamir) [reply ocs_ci/ocs/resources/noobaa.py]: Our main region is us-east-2 (Ohio).

Comment (ebenahar) [reply ocs_ci/ocs/resources/noobaa.py]: @RazTamir right!

Comment (ebenahar) [reply ocs_ci/ocs/resources/noobaa.py]: fixed

Comment (ebenahar) [reply ocs_ci/ocs/resources/noobaa.py]: need to re-set to region=config.ENV_DATA['region']

**Comment by MeridianExplorer** [comment]
Verified locally because NooBaa isn't deployed yet

========================== 3 passed in 237.39 seconds ==========================

**Pull request merged**